### PR TITLE
Port ANCM current to github.

### PR DESCRIPTION
**Pull request by jkotalik**
- Bring current ANCM on VSTS as a separate project in IISIntegration.
- Drops ANCM into two separate folders in the artifact folder: AspNetCoreModuleV1 and V2

TODO:
- [x] Re-add tests in IISIntegration, ServerTests, and MusicStore for current ANCM.
- [ ] Verify newly built ANCM doesn't cause regressions.
- [ ] Fix versioning of ANCM current
- [ ] Rename ANCM vNext aspnetcore.dll to aspnetcoreV2.dll.
- [ ] Freb logs? Websocket changes?

When reviewing this PR, I would recommend checking each commit rather than reviewing everything.

### CODE REVIEW

**Comment by jkotalik** [comment]
cc @muratg @Eilon

**Comment by Eilon** [comment]
Nice! But I don't know what to review ☹️

**Comment by jkotalik** [comment]
Nvrm, I was going to ask about the licenses, but it is the same as before.

**Comment by pakrym** [comment]
@muratg  No, lets squash some commits and review it first.

**Comment by muratg** [comment]
@pakrym didn't you guys want to take it as is first so that all the history is recorded?

**Comment by pakrym** [comment]
This PR contains both taking and fixing. @jkotalik already squashed it into 4 nice commits (there were 18 earlier today), I'll review them tomorrow and we'll merge without squashing to preserve history.

**Comment by pan-wang** [comment]
Nice work! The code is identical to IIS OOB now. I remember we updated couple of files to fix the issue introduced by the new compiler . Those changes are missing now. We should add them back.

**Comment by jkotalik** [comment]
> (there were 18 earlier today)
Shhhh they don't know that <_<

**Comment by Tratcher** [comment]
And this is happening side-by-side rather than in a separate branch because...?

**Comment by pakrym** [comment]
There is a bunch of ignored files in this PR. `src\AspNetCoreModuleV1\AspNetCore\aspnetcore_msg.rc`. `\src\AspNetCoreModuleV1\AspNetCore\ancm.mof` etc.

**Comment by jkotalik** [comment]
@Tratcher Few reasons

1. Easier to test ANCMV1 as it is part of dev and release/2.1
2. Building the MSIs can be done in the same build chain for both V1 and V2.

**Comment by Tratcher** [comment]
We don't have plans to build the ANCMv2 MSIs any time soon, correct? Their versioning would be all messed up.

**Comment by pakrym** [comment]
@shirhatti let's not do it as part of this PR. It's pretty large and impactful already.

File: .gitignore [hunk 1]
```diff
@@ -49,12 +49,13 @@ x64/
...
-src/RequestHandler/version.h
-src/CommonLib/aspnetcore_msg.h
-src/CommonLib/aspnetcore_msg.rc
+src/*/AspNetCore/aspnetcoremodule.h
+src/*/AspNetCore/aspnetcore_msg.h
+src/*/AspNetCore/aspnetcore_msg.rc
+src/*/*/version.h
+src/*/RequestHandler/version.h
+src/*/CommonLib/aspnetcore_msg.h
+src/*/CommonLib/aspnetcore_msg.rc
```
Comment (jkotalik) [inline .gitignore]: @pan-wang which of the ANCM .rc, .h, and .mof files need to be ignored?

File: src/AspNetCoreModuleV1/AspNetCore/src/precomp.hxx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+#pragma once
+#pragma warning( disable : 4091)
+
+//
+// System related headers
+//
+#define _WINSOCKAPI_
+
+#define NTDDI_VERSION 0x06010000
```
Comment (jkotalik) [inline src/AspNetCoreModuleV1/AspNetCore/src/precomp.hxx]: @pan-wang had to change this to do the ifdef for websockets.

**Comment by jkotalik** [comment]
I think this is good for review now. I have updated the ANCM out of process tests to select the version of the aspnetcore.dll version via a msbuild property. There are a few features we need to port from the ANCMV2 out of process code to V1 (https://github.com/aspnet/IISIntegration/commit/00b1948937e8d5cdcc7559c8d8aafda1613479b4, for example). 

For cross repo consumption of the ANCMV1, I created a separate nuget package called Microsoft.AspNetCore.AspNetCoreModuleV1. I will make a PR to universe to mark that as a noship nuget package for now. I will eventually update Microsoft.AspNetCore.AspNetCoreModule=> AspNetCoreModuleV2, but I will do that in a separate PR.

File: src/AspNetCoreModuleV1/AspNetCore/Inc/aspnetcore_msg.h [hunk 3]
```diff
@@ -0,0 +1,127 @@
+/*++
+
```
Comment (pan-wang) [inline src/AspNetCoreModuleV1/AspNetCore/Inc/aspnetcore_msg.h]: this file is auto generated and should not be included as source

File: src/AspNetCoreModuleV1/AspNetCore/Inc/forwardinghandler.h [hunk 4]
```diff
@@ -0,0 +1,479 @@
+#pragma once
+
+/*++
```
Comment (pan-wang) [inline src/AspNetCoreModuleV1/AspNetCore/Inc/forwardinghandler.h]: this needs to be updated with license info

File: src/AspNetCoreModuleV1/AspNetCore/AspNetCore.vcxproj [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+      <OptimizeReferences>true</OptimizeReferences>
+      <ModuleDefinitionFile>Source.def</ModuleDefinitionFile>
+      <AdditionalDependencies>kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;ahadmin.lib;rpcrt4.lib;winhttp.lib;pdh.lib;ws2_32.lib;wbemuuid.lib;iphlpapi.lib;%(AdditionalDependencies)</AdditionalDependencies>
+    </Link>
+    <ResourceCompile>
+      <AdditionalIncludeDirectories>..\Commonlib</AdditionalIncludeDirectories>
+    </ResourceCompile>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <None Include="ancm.mof" />
```
Comment (pan-wang) [inline src/AspNetCoreModuleV1/AspNetCore/AspNetCore.vcxproj]: will we ship FREB with this version of ANCM? if not, we don't need .mof file

File: src/AspNetCoreModuleV1/AspNetCore/Inc/resource.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (pan-wang) [inline src/AspNetCoreModuleV1/AspNetCore/Inc/resource.h]: we need add license info

File: src/AspNetCoreModuleV1/AspNetCore/AspNetCore.vcxproj (hunk 5 above)
Comment (shirhatti) [reply src/AspNetCoreModuleV1/AspNetCore/AspNetCore.vcxproj]: Why wouldn't we ship with FREB?

File: src/AspNetCoreModuleV1/AspNetCore/AspNetCore.vcxproj [hunk 7]
```diff
@@ -279,4 +272,22 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
+
+  <ItemGroup>
```
Comment (pakrym) [inline src/AspNetCoreModuleV1/AspNetCore/AspNetCore.vcxproj]: Remove Condition from one set of elements and remove everything else.

**Comment by pan-wang** [comment]
Yes.  Please merge the pr

**Pull request merged**